src/script-child-process.ts

```diff
@@ -119,10 +188,74 @@ export class ScriptChildProcess {
         } else {
           resolve({ok: true, value: undefined});
         }
+        this.#state = 'stopped';
       });
     });
   }
 
+  /**
+   * Mark this child process for termination. On Linux/macOS, sends a `SIGINT`
+   * signal. On Windows, invokes `taskkill /pid PID /t /f`.
+   *
+   * Note this function returns immediately. To find out when the process
+   * actually terminates, use the {@link completed} promise.
+   */
+  terminate(): void {
+    switch (this.#state) {
+      case 'started': {
+        this.#actuallyTerminate();
+        return;
+      }
+      case 'starting': {
+        // We're still starting up, and it's not possible to abort. When we get
+        // the "spawn" event, we'll notice the "stopping" state and actually
+        // terminate then.
+        this.#state = 'stopping';
+        return;
+      }
+      case 'stopping':
+      case 'stopped': {
+        // No-op.
+        return;
+      }
+      default: {
+        const never: never = this.#state;
+        throw new Error(
+          `Internal error: unexpected ScriptChildProcessState: ${String(never)}`
+        );
+      }
+    }
+  }
+
+  #actuallyTerminate(): void {
+    if (this.#child.pid === undefined) {
+      throw new Error(
+        `Internal error: Can't terminate child process because it has no pid. ` +
+          `Command: ${JSON.stringify(this.#script.command)}.`
+      );
+    }
+    if (IS_WINDOWS) {
+      // Windows doesn't have signals. Node ChildProcess.kill() sort of emulates
+      // the behavior of SIGKILL (and ignores the signal you pass in), but this
+      // doesn't end child processes. We have child processes because the parent
+      // process is cmd.exe or PowerShell.
+      // https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/taskkill
+      spawn('taskkill', [
+        '/pid',
+        this.#child.pid.toString(),
+        /* End child processes */ '/t',
+        /* Forceful */ '/f',
```

Comment:
Sounds reasonable, done, and will see if the tests pass.

(In the future we may want to support harder kills in some situations, e.g. if the user Ctrl-C's twice in a row. So then we'd do `\f` on Windows, and `SIGTERM` on posix.)

Reply:
Update, in https://github.com/google/wireit/pull/220, removing the `/f` flag prevented Wireit from actually killing the Node process we spawn in the tests. I can't say I understand why -- I suppose it could be that Node doesn't actually respond to the Windows kill signal in the same way it does `SIGINT` on posix -- but I think that means we need `/f` to be reliable.